java/lang/security/documentbuilderfactory-disallow-doctype-decl-false.yaml

```diff
@@ -0,0 +1,47 @@
+rules:
+  - id: documentbuilderfactory-disallow-doctype-decl-false
+    severity: ERROR
+    metadata:
+      cwe: "CWE-611: Improper Restriction of XML External Entity Reference"
+      owasp: "A4: XML External Entities (XXE)"
+      asvs:
+        section: V5 Validation, Sanitization and Encoding
+        control_id: 5.5.2 Insecue XML Deserialization
+        control_url: https://github.com/OWASP/ASVS/blob/master/4.0/en/0x13-V5-Validation-Sanitization-Encoding.md#v55-deserialization-prevention
+        version: "4"
+      references:
+        - https://blog.sonarsource.com/secure-xml-processor
+        - https://xerces.apache.org/xerces2-j/features.html
+      category: security
+      technology:
+        - java
+    message: >-
+      DOCTYPE declaratons are enabled for this DocumentBuilderFactory.
+      Without prohibiting external entity declarations, this is vulnerable to XML external entity attacks. 
+      Disable this by setting the feature "http://apache.org/xml/features/disallow-doctype-decl" to true.
+      Alternatively, allow DOCTYPE declarations and only prohibit external entities declarations.
+      This can be done by setting the features "http://xml.org/sax/features/external-general-entities" and "http://xml.org/sax/features/external-parameter-entities" to false.
+    patterns:
+      - pattern: $DBFACTORY.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false);
+      - pattern-not-inside: |
+          $RETURNTYPE $METHOD(...){
+            ...
+            $DBF.setFeature("http://xml.org/sax/features/external-general-entities", ...);
+            ...
+            $DBF.setFeature("http://xml.org/sax/features/external-parameter-entities", ...);
+            ...
+          }
+      - pattern-not-inside: |
+          $RETURNTYPE $METHOD(...){
+            ...
+            $DBF.setFeature("http://xml.org/sax/features/external-parameter-entities", ...);
+            ...
+            $DBF.setFeature("http://xml.org/sax/features/external-general-entities", ...);
+            ...
+          }
+    fix: |
+      $DBFACTORY.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false);
```

Comment:
seems like this fix isn't working perfectly 😭 : https://semgrep.dev/s/A5w0. Definitely a problem with the autofix engine itself, but perhaps we can skip the fix for now and file an issue?